DataDog/datadog-agent
# dispatch ad templates for endpoints

### What does this PR do?

make the DCA watch services and endpoints, and create config template for endpoints of services that have AD annotations attached, serve them on `api/v1/endpointschecks/config/{nodename}`.

add new config provider named `endpointschecks` to consume these templates and tag them with pod's tags

### Motivation

make k8s service check more precise by performing checks on the endpoints (in the node level)

### QA

- doc https://github.com/DataDog/documentation/pull/4412

- try to enable/disable/modify the checks by changing the annotations and see if we get the expected result

- try multiple services together with their endpoints checks enabled

- try to enable endpointschecks only, clusterchecks only and both (as config providers on the node agent)

node agent image needed to test the feature:
`datadog/agent-dev:ahmed-mez-dispatch-ad-templates-for-eps`
